pkg/driver/controller.go

```diff
@@ -162,8 +167,28 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		basePath = value
 	}
 
+	if value, ok := volumeParams[AzName]; ok {
+		azName = value
+	}
+
+	// Fetch aws role ARN for cross account mount from secrets
+	secrets := req.GetSecrets()
+
+	if value, ok := secrets[RoleArn]; ok {
+		roleArn = value
+	}
+
+	if roleArn != "" {
+		localCloud, err = cloud.NewCloudWithRole(roleArn)
+		if err != nil {
+			return nil, status.Errorf(codes.Unauthenticated, "Unable to initialize aws cloud: %v. Please verify role has the correct AWS permissions for cross account mount", err)
```

Comment:
Where should the roleArn be configured? in .config file?

Reply:
The roleArn should be provided via secret

Reply:
What is secret? Is that some specific configuration for csi-driver?

Reply:
Secret is an object which lets you store information, in this case we rely on users to configure roleArn through secrets.

K8s allows secrets to be accessed by a pod in multiple ways. In our context, if a user provides inputs secret names & namespace into the storage class, [external-provisioner](https://kubernetes-csi.github.io/docs/external-provisioner.html) will extract it and pass it on to csi-driver's Controller service `rpc`, provided it has right RBAC rule to access the secret

More info on how secrets work for CSI operations: https://kubernetes-csi.github.io/docs/secrets-and-credentials.html#csi-driver-secrets